# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalResponse.h
# Commit: 4fd92351d64b
# Full Hash: 4fd92351d64b1cbeb82beaf6228b21c01dc9bc03
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - FetchService integration for PFetch. r=dom-worker-reviewers,jesup
#   
#   Depends on D142436
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142437
# ==============================================================================

diff -r 7a4e3f5f674a -r 4fd92351d64b dom/fetch/InternalResponse.h
--- a/dom/fetch/InternalResponse.h	Wed Jan 18 00:50:19 2023 +0000
+++ b/dom/fetch/InternalResponse.h	Wed Jan 18 00:50:20 2023 +0000
@@ -311,6 +311,9 @@
 
   bool HasBeenCloned() const { return mCloned; }
 
+  void SetSerializeAsLazy(bool aAllow) { mSerializeAsLazy = aAllow; }
+  bool CanSerializeAsLazy() const { return mSerializeAsLazy; }
+
   void InitChannelInfo(nsIChannel* aChannel) {
     mChannelInfo.InitFromChannel(aChannel);
   }
@@ -342,6 +345,13 @@
 
   InternalResponseMetadata GetMetadata();
 
+  RequestCredentials GetCredentialsMode() const {
+    if (mWrappedResponse) {
+      return mWrappedResponse->GetCredentialsMode();
+    }
+    return mCredentialsMode;
+  }
+
   ~InternalResponse();
 
  private:
@@ -380,6 +390,7 @@
   nsCOMPtr<nsIInputStream> mAlternativeBody;
   nsMainThreadPtrHandle<nsICacheInfoChannel> mCacheInfoChannel;
   bool mCloned;
+  bool mSerializeAsLazy{true};
 
  public:
   static constexpr int64_t UNKNOWN_BODY_SIZE = -1;